src/lib/sedlexing.ml

```diff
@@ -250,21 +311,12 @@ module Utf8 = struct
   module Helper = struct
     (* http://www.faqs.org/rfcs/rfc3629.html *)
 
-    let width = Array.make 256 (-1)
-
-    let () =
-      for i = 0 to 127 do
-        width.(i) <- 1
-      done;
-      for i = 192 to 223 do
-        width.(i) <- 2
-      done;
-      for i = 224 to 239 do
-        width.(i) <- 3
-      done;
-      for i = 240 to 247 do
-        width.(i) <- 4
-      done
+    let width = function
+      | '\000' .. '\127' -> 1
+      | '\192' .. '\223' -> 2
+      | '\224' .. '\239' -> 3
+      | '\240' .. '\247' -> 4
+      | _ -> -1
```

Comment:
Why not `raise Malformed` here directly? Performance?

Reply:
Why not ! I've pushed the change.